test/NormalModule.test.js

```diff
@@ -0,0 +1,410 @@
+/* globals describe, it, beforeEach, afterEach */
+"use strict";
+require("should");
+const sinon = require("sinon");
+const NormalModule = require("../lib/NormalModule");
+const path = require("path");
+const SourceMapSource = require("webpack-sources").SourceMapSource;
+const OriginalSource = require("webpack-sources").OriginalSource;
+const RawSource = require("webpack-sources").RawSource;
+
+describe("NormalModule", function() {
+	let normalModule;
+	let request;
+	let userRequest;
+	let rawRequest;
+	let loaders;
+	let resource;
+	let parser;
+	beforeEach(function() {
+		request = "some/request";
+		userRequest = "some/userRequest";
+		rawRequest = "some/rawRequest";
+		loaders = [];
+		resource = "some/resource";
+		parser = {
+			parser() {}
```

Comment:
This is weird. A Parser doesn't have a parser method. I would expect `constructor() {}` or `parse() {}`.